clustering/operations.go

```diff
@@ -149,14 +149,36 @@ func (p *managerProcess) clone(ctx context.Context, ss *StatusSet) (bool, error)
 func (p *managerProcess) switchover(ctx context.Context, ss *StatusSet) error {
 	log := logFromContext(ctx)
 	log.Info("begin switchover the primary", "current", ss.Primary, "next", ss.Candidate)
-
 	pdb := ss.DBOps[ss.Primary]
-	if err := pdb.SetReadOnly(ctx, true); err != nil {
-		return fmt.Errorf("failed to make instance %d read-only: %w", ss.Primary, err)
+
+	// SetReadOnly waits for a running DML.
+	// Therefore, if it waits for a long time, deleteGracePeriodSeconds may be reached.
+	// To avoid this, set a timeout on SetReadOnly.
+	// If SetReadOnly fails, kill all processes and retry.
+	readOnly := false
+	for i := 0; i < 2; i++ {
+		ctxTimeout, cancel := context.WithTimeout(ctx, 15*time.Second)
```

Comment:
@shunki-fujita 
Can we use `SET SESSION wait_timeout` instead?
https://dev.mysql.com/doc/refman/8.4/en/server-system-variables.html#sysvar_wait_timeout

Reply:
@ymmt2005 
Using `lock_wait_timeout` sounds like a good idea.
Thanks.

```
mysql> SET GLOBAL super_read_only=1;
ERROR 1205 (HY000): Lock wait timeout exceeded; try restarting transaction
mysql>
```